packages/flutter_tools/lib/src/commands/screenshot.dart

```diff
@@ -14,22 +14,39 @@ import '../runner/flutter_command.dart';
 import '../vmservice.dart';
 
 const String _kOut = 'out';
-const String _kSkia = 'skia';
+const String _kType = 'type';
+const String _kObservatoryPort = 'observatory-port';
+const String _kDeviceType = 'device';
+const String _kSkiaType = 'skia';
+const String _kRasterizerType = 'rasterizer';
 
 class ScreenshotCommand extends FlutterCommand {
   ScreenshotCommand() {
     argParser.addOption(
       _kOut,
       abbr: 'o',
+      valueHelp: 'path/to/file',
       help: 'Location to write the screenshot.',
     );
     argParser.addOption(
-      _kSkia,
+      _kObservatoryPort,
       valueHelp: 'port',
-      help: 'Retrieve the last frame rendered by a Flutter app as a Skia picture\n'
-        'using the specified observatory port.\n'
-        'To find the observatory port number, use "flutter run --verbose"\n'
-        'and look for "Forwarded host port ... for Observatory" in the output.'
+      help: 'The observatory port to connect to.\n'
+          'This is required when --$_kType is "$_kSkiaType" or "$_kRasterizerType".\n'
+          'To find the observatory port number, use "flutter run --verbose"\n'
+          'and look for "Forwarded host port ... for Observatory" in the output.',
+    );
+    argParser.addOption(
+      _kType,
+      valueHelp: 'type',
+      help: 'The type of screenshot to retrieve.',
+      allowed: const <String>[_kDeviceType, _kSkiaType, _kRasterizerType],
+      allowedHelp: const <String, String>{
+        _kDeviceType: 'Delegate to the device\'s native screenshot capabilities.',
```

Comment:
"Screenshot the entire screen currently being displayed (includes contents not rendered by Flutter)"

Reply:
Done